src/wallet/rpcdump.cpp

```diff
@@ -246,6 +334,82 @@ UniValue importaddress(const UniValue& params, bool fHelp)
     return NullUniValue;
 }
 
+UniValue importaddresses(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 1 || params.size() > 4)
+        throw runtime_error(
+            "importaddresses [rescan | no-rescan] \"address\"...\n"
+            "\nAdds a script (in hex) or address that can be watched as if it were in your wallet but cannot be used to spend.\n"
+            "\nArguments:\n"
+            "1. \"rescan | no-rescan\" (string, optional default rescan) If \"no-rescan\", skip wallet rescan\n"
+            "1. \"address\"           (string, 0 or more) The address or hex-encoded script\n"
+            "\nNote, this command will return before the rescan (may take hours) is complete..\n"
+            "If you have the full public key, you should call importpublickey instead of this.\n"
+            "\nExamples:\n"
+            "\nImport a script with rescan\n"
+            + HelpExampleCli("importaddresses", "\"myscript\"") +
+            "\nImport using a label without rescan\n"
+            + HelpExampleCli("importaddresses", "no-rescan \"myscript\"") +
+            "\nAs a JSON-RPC call\n"
+            + HelpExampleRpc("importaddress", "\"myscript\"")
+        );
+
+    string strLabel = "";
+    if (params.size() > 1)
+        strLabel = params[1].get_str();
+
+    // Whether to perform rescan after import
+    bool fRescan = true;
+    // Whether to import a p2sh version, too
+    bool fP2SH = false;
+
+    unsigned int paramNum = 0;
+
+    if (params[0].get_str() == "no-rescan")
+    {
+        fRescan = false;
+        paramNum++;
+    }
+    else if (params[0].get_str() == "rescan")
+    {
+        fRescan = true;
+        paramNum++;
+    }
+
+    if (fRescan && fPruneMode)
+        throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled in pruned mode");
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    for (; paramNum < params.size(); paramNum++)
+    {
+        CBitcoinAddress address(params[paramNum].get_str());
+        if (address.IsValid())
```

Comment:
FYI, you will have merge conflicts when this gets pulled into the BitcoinCash branch because of CBitcoinAddress which no longer exists in the Cash branch.